src/Mvc/Mvc.Core/test/ModelBinding/Binders/DictionaryModelBinderTest.cs

```diff
@@ -111,8 +112,8 @@ public static TheoryData<string, string, IDictionary<string, string>> StringToSt
 
             return new TheoryData<string, string, IDictionary<string, string>>
                 {
-                    { string.Empty, "[{0}]", dictionaryWithOne },
-                    { string.Empty, "[{0}]", dictionaryWithThree },
+                    { string.Empty, "[{0}]", dictionaryWithNone },
```

Comment:
If prefix is empty, we should not be trying to bind all values into the dictionary.